resources/resource_cache.go

```diff
@@ -68,6 +68,10 @@ func (c *ResourceCache) Get(ctx context.Context, key string) (resource.Resource,
 }
 
 func (c *ResourceCache) GetOrCreate(key string, f func() (resource.Resource, error)) (resource.Resource, error) {
+	// Do nothing if resource is nil.
+	if r, _ := f(); r == nil {
```

Comment:
The main motivation behind this cache is hav `f` being called once, which this breaks.

I suggest that we instead fix this by adding a prefix to the cache key in `Copy` (and also do the same in `Get` to make it symmetrical).

```go
func (c *Client) Copy(r resource.Resource, targetPath string) (resource.Resource, error) {
	key :=  dynacache.CleanKey(targetPath) + "__copy"
	return c.rs.ResourceCache.GetOrCreate(key, func() (resource.Resource, error) {
		return resources.Copy(r, targetPath), nil
	})
}

```

Or something.

Reply:
Added suffix to Get and Copy, and added test case for https://github.com/gohugoio/hugo/issues/10412.